### PR TITLE
feat: Add once for each item support for JS task runner (no-changelog)

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Make the JS task runner work with the 'runOnceForEachItem' execution mode. A single request is used for the `runOnceForEachItem` mode and do the iteration on the task runner. This makes the JS task runner more coupled to the Code Node, but I think that's a fair tradeoff as that's its main purpose. The performance of this approach is almost 3x compared to the performance of using a separate request for each iteration in the JS Code benchmark.

This also makes the JS task runner output match 1:1 with the old sandbox approach: errors are handled correctly (continue on fail), items are normalized and validated, `item` and `items` aliases are provided in the execution context.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/PAY-2042/use-task-runners-for-code-nodes-runonceforeachitem
https://linear.app/n8n/issue/PAY-2058/more-optimized-version-of-runonceforeachitem

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/task-runner/src/execution-error.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { ApplicationError } from 'n8n-workflow';
```
Comment (tomi) [inline packages/@n8n/task-runner/src/execution-error.ts:1]: Copied from `nodes-base` as is

File: packages/@n8n/task-runner/src/execution-error.ts
[the remark this answers is not in the document]
Comment (valya) [reply packages/@n8n/task-runner/src/execution-error.ts:1]: There's a decent amount of duplication in the `task-runner` package (mostly mine :see_no_evil:), we should probably circle back later and deal with it.

**Review by valya — APPROVED** [review]

File: packages/@n8n/task-runner/src/execution-error.ts
[the remark this answers is not in the document]
Comment (tomi) [reply packages/@n8n/task-runner/src/execution-error.ts:1]: Yeah hopefully we can delete this code from `nodes-base` once we have the task runners stable

**Comment by janober** [comment]
Got released with [`n8n@1.63.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.63.0)

**Pull request merged**